code/modules/power/turbine.dm

```diff
@@ -0,0 +1,431 @@
+	name = "gas turbine generator"
+	desc = "A gas turbine used for backup power generation."
+	icon = 'icons/obj/pipeturbine.dmi'
+	icon_state = "turbine"
+	anchored = TRUE
+	density = TRUE
+	var/opened = FALSE
+	var/obj/machinery/power/compressor/compressor
+	var/turf/simulated/outturf
+	var/lastgen
+	/// If the turbine is outputing enough to visibly affect its sprite
+	var/generator_threshold = FALSE
+	var/productivity = 1
+	component_types = list(
+		/obj/item/stock_parts/capacitor = 6,
+		/obj/item/stack/cable_coil = 5
+	)
+
+/obj/machinery/computer/terminal/turbine_computer
+	name = "gas turbine control computer"
+	desc = "A computer to remotely control a gas turbine. Link it to a turbine via use of a multitool."
+	icon_screen = "turbinecomp"
+	icon_keyboard = "tech_key"
+	var/obj/machinery/power/compressor/compressor
+	var/list/obj/machinery/door/blast/doors
+	var/id = 0
+	var/door_status = 0
+
+// the inlet stage of the gas turbine electricity generator
+
+/obj/machinery/power/compressor/Initialize(mapload)
+	. = ..()
+// The inlet of the compressor is the direction it faces
+	gas_contained = new
+	inturf = get_step(src, dir)
+	locate_machinery()
+	if(!turbine)
+		stat |= BROKEN
+	else
+		turbine.compressor = src
+
+	//Radio for screaming about overheats
+	radio = new(src)
+	radio.set_listening(FALSE)
+	radio.config(list("Engineering" = 0))
+
+#define COMPFRICTION 5e5
+#define COMPSTARTERLOAD 2800
+
+
+// Crucial to make things work!!!!
+// OLD FIX - explanation given down below.
+// /obj/machinery/power/compressor/CanPass(atom/movable/mover, turf/target, height=0)
+// 		return !density
+
+/obj/machinery/power/compressor/proc/locate_machinery()
+	if(turbine)
+		return
+	turbine = locate() in get_step(src, get_dir(inturf, src))
+	if(turbine)
+		turbine.locate_machinery()
+
+/obj/machinery/power/compressor/RefreshParts()
+	var/E = 0
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		E += M.rating
+	efficiency = E / 6
+
+/obj/machinery/power/compressor/attackby(obj/item/attacking_item, mob/user, params)
+	if(attacking_item.iswrench())
+		turbine = null
+		inturf = get_step(src, dir)
+		locate_machinery()
+		if(turbine)
+			to_chat(user, "<span class='notice'>Turbine connected.</span>")
+			stat &= ~BROKEN
+		else
+			to_chat(user, "<span class='alert'>Turbine not connected.</span>")
+			stat |= BROKEN
+		return
+
+	if(default_deconstruction_crowbar(user, attacking_item))
+		return TRUE
+
+	if(default_deconstruction_screwdriver(user, attacking_item))
+		return TRUE
+
+	return ..()
+
+/obj/machinery/power/compressor/proc/trigger_overheat()
+	starter = FALSE
+	last_overheat = world.time
+	overheat -= 50
+	radio.autosay(OVERHEAT_MESSAGE, name, "Engineering")
+	playsound(src, 'sound/machines/buzz-two.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
+
+/obj/machinery/power/compressor/proc/time_until_overheat_done()
+	return max(last_overheat + OVERHEAT_TIME - world.time, 0)
+
+/obj/machinery/power/compressor/process()
```

Comment:
process needs the parameter `ticks_per_seconds` as parent and to multiply everything related to time with it